services/src/kilda-pce/src/main/java/org/openkilda/pce/impl/AvailableNetwork.java

```diff
@@ -113,6 +118,43 @@ public void processDiversitySegments(List<PathSegment> segments) {
         }
     }
 
+    /**
+     * Adds diversity weights into {@link AvailableNetwork} based on passed path segments and configuration.
+     */
+    public void processDiversitySegmentsWithPop(List<PathSegment> segments) {
+        if (segments.size() <= 1) {
+            return;
+        }
+
+        Set<String> allocatedPopSet = new HashSet<>();
+
+        for (int i = 0; i < segments.size(); i++) {
+            PathSegment ps = segments.get(i);
+            if (i == 0) {
+                allocatedPopSet.add(ps.getDestSwitch().getPop());
+            } else if (i == segments.size() - 1) {
+                allocatedPopSet.add(ps.getSrcSwitch().getPop());
+            } else {
+                allocatedPopSet.add(ps.getSrcSwitch().getPop());
+                allocatedPopSet.add(ps.getDestSwitch().getPop());
+            }
+        }
+        for (Edge edge : edges) {
+            String srcPop = edge.getSrcSwitch().getPop();
+            if (srcPop != null && allocatedPopSet.contains(srcPop)) {
+                edge.increaseDiversityGroupPerPopUseCounter();
+                continue;
+            }
+            String dstPop = edge.getDestSwitch().getPop();
+            if (dstPop != null && allocatedPopSet.contains(dstPop)) {
+                edge.increaseDiversityGroupPerPopUseCounter();
+                continue;
```

Comment:
nit: this continue is redundant.

Reply:
it was added intentionally so in case of upcoming changes it won't be forgotten